src/platform/Linux/ThreadStackManagerImpl.cpp

```diff
@@ -685,15 +685,169 @@ CHIP_ERROR ThreadStackManagerImpl::_WriteThreadNetworkDiagnosticAttributeToTlv(A
+    case ThreadNetworkDiagnostics::Attributes::TxErrCcaCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::TxErrAbortCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::TxErrBusyChannelCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxTotalCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxUnicastCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxBroadcastCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxDataCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxDataPollCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxBeaconCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxBeaconRequestCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxOtherCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxAddressFilteredCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxDestAddrFilteredCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxDuplicatedCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrNoFrameCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrUnknownNeighborCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrInvalidSrcAddrCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrSecCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrFcsCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::RxErrOtherCount::Id:
+        err = encoder.Encode(static_cast<uint32_t>(0));
+        break;
+    case ThreadNetworkDiagnostics::Attributes::ActiveTimestamp::Id:
+        err = encoder.Encode(static_cast<uint64_t>(0));
```

Comment:
Why is this not null?  Seriously, please check this against the spec.  There are more things here that don't match....